flink-table/flink-table-common/src/main/java/org/apache/flink/table/descriptors/DescriptorProperties.java

```diff
@@ -582,19 +607,26 @@ public short getShort(String key) {
 		final TableSchema.Builder schemaBuilder = TableSchema.builder();
 		for (int i = 0; i < fieldCount; i++) {
 			final String nameKey = key + '.' + i + '.' + TABLE_SCHEMA_NAME;
-			final String typeKey = key + '.' + i + '.' + TABLE_SCHEMA_TYPE;
+			final String legacyTypeKey = key + '.' + i + '.' + TABLE_SCHEMA_TYPE;
+			final String typeKey = key + '.' + i + '.' + TABLE_SCHEMA_DATATYPE;
 			final String exprKey = key + '.' + i + '.' + TABLE_SCHEMA_EXPR;
 
 			final String name = optionalGet(nameKey).orElseThrow(exceptionSupplier(nameKey));
 
-			final TypeInformation<?> type = optionalGet(typeKey)
-				.map(TypeStringUtils::readTypeInfo)
-				.orElseThrow(exceptionSupplier(typeKey));
+			final DataType type;
+			if (containsKey(typeKey)) {
```

Comment:
NIT: use `? :`

Reply:
We can't. It is not `if else`, but a `if .. else if .. else`.